common/plugins/org.polarsys.capella.common.helpers/src/org/polarsys/capella/common/helpers/EObjectExt.java

```diff
@@ -38,16 +41,38 @@
  */
 public class EObjectExt extends EcoreUtil2 {
 
+  static CachedBiFunction<EObject, EClass, Set<EObject>> getAll_withEClass = MDEQueries.getInstance()
+      .getAllQueries()::getAll;
+  static CachedBiFunction<EObject, Predicate<EObject>, Set<EObject>> getAll_withPredicate = MDEQueries.getInstance()
+      .getAllQueries()::getAll;
+
   /**
    * @param currentElement
    * @param targetType
    * @return Set<EObject>
    */
   public static Set<EObject> getAll(EObject currentElement, EClass targetType) {
-    List<EClass> filters = new ArrayList<EClass>();
+    return new HashSet<EObject>(getAll_withEClass.get(currentElement, targetType));
```

Comment:
LinkedHashSet to keep order as defined in the original method. it was probably important

Reply:
Done